rdopkg/utils/specfile.py

```diff
@@ -245,6 +245,11 @@ def patches_apply_method(self):
             return 'autosetup'
         return 'rpm'
 
+    def set_commit_ref_macro(self, ref):
+        self._txt = re.sub(
+            r'^\%global commit \w+',
+            '%%global commit %s' % ref, self.txt, flags=re.M)
```

Comment:
Thanks @yac and @hguemar , I've amended the commit to make the regex changes, and added a test to the test suite to cover this behavior.